pydantic/_internal/_decorators.py

```diff
@@ -218,3 +256,127 @@ def in_ipython() -> bool:
         return False
     else:  # pragma: no cover
         return True
+
+
+class OnlyValueValidator(Protocol):
+    def __call__(self, __value: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithValues(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any]) -> Any:
+        ...
+
+
+class V1ValidatorWithValuesKwOnly(Protocol):
+    def __call__(self, __value: Any, *, values: dict[str, Any]) -> Any:
+        ...
+
+
+class V1ValidatorWithKwargs(Protocol):
+    def __call__(self, __value: Any, **kwargs: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithKwargsAndValue(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any], **kwargs: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithValuesAndKwargs(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any], **kwargs: Any) -> Any:
+        ...
+
+
+V1Validator = Union[
+    V1ValidatorWithValues, V1ValidatorWithValuesKwOnly, V1ValidatorWithKwargs, V1ValidatorWithValuesAndKwargs
+]
+
+
+def make_generic_validator(
+    validator: V1Validator | OnlyValueValidator | ValidatorFunction, mode: str
+) -> ValidatorFunction:
+    """
+    Make a generic function which calls a validator with the right arguments.
```

Comment:
I personally would appreciate a little more description about why this is necessary; I looked at the use site and it wasn't immediately obvious. (Not suggesting this function isn't necessary — just that it might help me, or others reading this code, get our bearings to have a little more context about what's happening.)

I think it could say something to the effect of:
```
In order to support different signatures, including deprecated validator signatures from v1,
we introspect the function signature and wrap it in a parent function that has a signature
compatible with pydantic_core
```
(Or something similar that is more detailed/accurate, if the above is wrong)

Reply:
Done